x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java

```diff
@@ -118,49 +120,7 @@ protected Object unwrapFieldsMultiValue(Object values) {
             return unwrapped;
         }
 
-        // The Jackson json parser can generate for numerics - Integers, Longs, BigIntegers (if Long is not enough)
-        // and BigDecimal (if Double is not enough)
-        if (values instanceof Number || values instanceof String || values instanceof Boolean) {
-            if (dataType == null) {
-                return values;
-            }
-            if (dataType.isNumeric() && isFromDocValuesOnly(dataType) == false) {
-                if (dataType == DataTypes.DOUBLE || dataType == DataTypes.FLOAT || dataType == DataTypes.HALF_FLOAT) {
-                    Number result = null;
-                    try {
-                        result = numberType(dataType).parse(values, true);
-                    } catch(IllegalArgumentException iae) {
-                        return null;
-                    }
-                    // docvalue_fields is always returning a Double value even if the underlying floating point data type is not Double
-                    // even if we don't extract from docvalue_fields anymore, the behavior should be consistent
-                    return result.doubleValue();
-                } else {
-                    Number result = null;
-                    try {
-                        result = numberType(dataType).parse(values, true);
-                    } catch(IllegalArgumentException iae) {
-                        return null;
-                    }
-                    return result;
-                }
-            } else if (DataTypes.isString(dataType) || dataType == DataTypes.IP) {
-                return values.toString();
-            } else {
-                return values;
-            }
-        }
-        throw new QlIllegalArgumentException("Type {} (returned by [{}]) is not supported", values.getClass().getSimpleName(), fieldName);
-    }
-
-    protected boolean isFromDocValuesOnly(DataType dataType) {
-        return dataType == KEYWORD // because of ignore_above.
-                    || dataType == DATETIME
-                    || dataType == SCALED_FLOAT; // because of scaling_factor
-    }
-    
-    private static NumberType numberType(DataType dataType) {
-        return NumberType.valueOf(dataType.esType().toUpperCase(Locale.ROOT));
+        return values;
```

Comment:
Isn't the left side removed due to the use of fields API? Just double checking why the change appears to belong to this PR in Github.

Reply:
It was a merge mistake, it's reverted already.